src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/notifications/ReferralNotificationService.kt

```diff
@@ -56,6 +65,36 @@ class ReferralNotificationService(
           )
         )
       }
+
+      ReferralEventType.DETAILS_CHANGED -> {
+        handleDetailsChangedEvent(event)
+      }
+    }
+  }
+
+  private fun handleDetailsChangedEvent(event: ReferralEvent) {
+    val newDetails = event.data["newDetails"] as ReferralDetailsDTO
+    val previousDetails = event.data["previousDetails"] as ReferralDetailsDTO
+
+    val recipient = hmppsAuthService.getUserDetail(event.data["recipient"] as AuthUserDTO)
+    val updater = hmppsAuthService.getUserDetail(authUserRepository.findByIdOrNull(newDetails.createdById)!!)
```

Comment:
i don't like having to make the repository call here - any thoughts?

Reply:
Is it possible to populate that in ReferralDetailsDTO itself?

Reply:
To me, making lookups based on an absolute value (primary key) is fine and reasonable

I would probably have concerns about looking up something that may be a "dirty read" -- eg. a publish that happened during a transaction and the listener code kicks off before the transaction commits